### PR TITLE
Added prayer plugin for kotlin

### PR DESCRIPTION
**Pull request by tlf30**
Untested. I think it works. Would love if someone could test it. 

~Trevor

### CODE REVIEW

File: game/plugin/skills/prayer/src/data.kt [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    INCREDIBLE_REFLEXES(5620, 34, 94, 0.1),
+    PROTECT_FROM_MAGIC(5621, 37, 95, 0.15),
+    PROTECT_FROM_MISSILES(5622, 40, 96, 0.15),
+    PROTECT_FROM_MELEE(5623, 43, 97, 0.15),
+    RETRIBUTION(683, 46, 98, 0.15),
+    REDEMPTION(684, 49, 99, 0.15),
+    SMITE(685, 52, 100, 0.2);
+
+    companion object {
+        fun findByButton(button: Int): Prayer? = Prayer.values().find { prayer -> prayer.button == button }
```
Comment (garyttierney) [inline game/plugin/skills/prayer/src/data.kt]: `Enum#values()` returns a copy of the array every time.  Keeping a local copy of `Prayer.values` would be better.

File: game/plugin/skills/prayer/src/data.kt [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    MONKEY_BONES_LARGE_0(3181, 14.0),
+    MONKEY_BONES_LARGE_1(3182, 14.0),
+    MONKEY_BONES_SMALL_1(3183, 14.0),
+    SHAKING_BONES(3187, 14.0),
+    FAYRG_BONES(4830, 84.0),
+    RAURG_BONES(4832, 96.0),
+    OURG_BONES(4834, 140.0);
+
+    companion object {
+        fun findById(id: Int): Bone? = Bone.values().find { bone -> bone.id == id }
```
Comment (garyttierney) [inline game/plugin/skills/prayer/src/data.kt]: Same as Prayer.

File: game/plugin/skills/prayer/src/data.kt [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+    companion object {
+        fun findByButton(button: Int): Prayer? = Prayer.values().find { prayer -> prayer.button == button }
+    }
+}
+
+fun updatePrayer(player: Player, prayer: Prayer?) {
+    //Clear active prayer
+    if (prayer != null) {
+        if (player.getCurrentPrayer() == prayer) {
```
Comment (garyttierney) [inline game/plugin/skills/prayer/src/data.kt]: Players can have multiple Prayers active, I think.  An `EnumSet` would be a better way to store active ones.

**Review by garyttierney — DISMISSED** [review]
Again, mostly LGTM with a few nitpicks.

**Comment by tlf30** [comment]
How does that look @garyttierney??

File: game/plugin/skills/prayer/src/data.kt [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+import org.apollo.game.model.Animation
+import org.apollo.game.model.entity.Mob
+import org.apollo.game.model.entity.Player
+
+val BURY_BONE_ANIMATION = Animation(827)
+
+val PLAYER_PRAYERS = hashMapOf<Player, ArrayList<Prayer>?>()
+
+fun Player.getCurrentPrayers(): ArrayList<Prayer> {
+    PLAYER_PRAYERS[this] = PLAYER_PRAYERS[this] ?: ArrayList<Prayer>(); //Make sure that the array list exists
```
Comment (garyttierney) [inline game/plugin/skills/prayer/src/data.kt]: Any thoughts on a more generic way of handling these scenarios @Major- ?

Comment (Major-) [reply game/plugin/skills/prayer/src/data.kt]: Ye we need some sort of replacement for the attributes system, this is fine for now.

Although I think it should be a `Set` not `ArrayList` - we don't need ordering, right?

Comment (tlf30) [reply game/plugin/skills/prayer/src/data.kt]: Good catch, yes it does not need ordering.

**Pull request merged**